app/workers/sphinx_indexation_worker.rb

```diff
@@ -19,24 +19,6 @@ class SphinxIndexationWorker < ApplicationJob
   end
 
   def perform(model, id)
-    instance = model.find_by(model.primary_key => id)
-
-    if instance
-      reindex(instance)
-    else
-      delete_from_index(model, id)
-    end
-  end
-
-  protected
-
-  def reindex(instance)
-    ThinkingSphinx::Processor.new(instance: instance).upsert
-  end
-
-  def delete_from_index(model, *ids)
-    ids.each do |id|
-      ThinkingSphinx::Processor.new(model: model, id: id).delete
-    end
+    ThinkingSphinx::Processor.new(model: model, id: id).stage
```

Comment:
What does this do and how it replaces the old code?

Reply:
See the `stage` method implementation. It implements this upsert/delete in a better way, based on index scope, not only on the presence of the object.